src/js/components/FileInput/FileInput.js

```diff
@@ -0,0 +1,257 @@
+  &:focus {
+    ${focusStyle}
+  }
+  ${props => props.theme.fileInput && props.theme.fileInput.extend};
+  ${props =>
+    props.hover &&
+    props.theme.fileInput &&
+    props.theme.fileInput.hover &&
+    props.theme.fileInput.hover.extend};
+  ${props =>
+    props.dragOver &&
+    props.theme.fileInput &&
+    props.theme.fileInput.dragOver &&
+    props.theme.fileInput.dragOver.extend};
+`;
+
+const Label = styled(Text)`
+  ${props =>
+    props.theme.fileInput &&
+    props.theme.fileInput.label &&
+    props.theme.fileInput.label.extend};
+`;
+
+const Message = styled(Text)`
+  ${props =>
+    props.theme.fileInput &&
+    props.theme.fileInput.message &&
+    props.theme.fileInput.message.extend};
+`;
+
+const FileInput = forwardRef(
+  (
+    {
+      a11yTitle,
+      background,
+      border,
+      disabled,
+      id,
+      renderFile,
+      messages,
+      margin,
+      multiple,
+      name,
+      onChange,
+      pad,
+      value: valueProp,
+      ...rest
+    },
+    ref,
+  ) => {
+    const theme = useContext(ThemeContext);
+    const formContext = useContext(FormContext);
+    const [files, setFiles] = formContext.useFormInput(name, valueProp, []);
+    const [hover, setHover] = React.useState();
+    const [dragOver, setDragOver] = React.useState();
+    const aggregateThreshold = (multiple && multiple.aggregateThreshold) || 10;
+    const inputRef = useForwardedRef(ref);
+    const controlRef = useRef();
+    const removeRef = useRef();
+    const RemoveIcon = theme.fileInput.icons.remove;
+
+    const mergeTheme = (propertyName, defaultKey) => {
+      let result = {};
+      const themeProp = theme.fileInput[propertyName];
+      if (themeProp)
+        if (typeof themeProp !== 'object')
+          if (defaultKey) result[defaultKey] = themeProp;
+          else result = themeProp;
+        else result = { ...themeProp };
+      const hoverThemeProp = theme.fileInput.hover[propertyName];
+      if (hover && hoverThemeProp)
+        if (typeof hoverThemeProp !== 'object')
+          if (defaultKey) result[defaultKey] = hoverThemeProp;
+          else result = hoverThemeProp;
+        else result = { ...result, ...hoverThemeProp };
+      const dragOverThemeProp = theme.fileInput.dragOver[propertyName];
+      if (dragOver && dragOverThemeProp)
+        if (typeof dragOverThemeProp !== 'object')
+          if (defaultKey) result[defaultKey] = dragOverThemeProp;
+          else result = dragOverThemeProp;
+        else result = { ...result, ...dragOverThemeProp };
+      return typeof result === 'object' && Object.keys(result).length === 0
+        ? undefined
+        : result;
+    };
+
+    return (
+      <Keyboard
+        onSpace={event => {
+          if (controlRef.current === event.target) inputRef.current.click();
+        }}
+        onEnter={event => {
+          if (controlRef.current === event.target) inputRef.current.click();
+        }}
+      >
+        <ContentsBox
```

Comment:
Not getting any focus indication when I tab to the input

Reply:
Also, when testing with VoiceOver, in the scenario with multiple files already in the input, if I tab to the input again it focuses on the "X" button of a file. I can't open the file browser or get focus on it until I delete a file (after which focus gets put on the input container).

seems like we would want users to be able to upload more files at any stage of the process (no files uploaded yet or already uploaded files)

Reply:
good catches! Should be fixed now.